examples/asana-googlecalendar/index.js

```diff
@@ -0,0 +1,183 @@
+    return res.redirect("/tasks");
+  }
+
+  // If no userId in session, redirect to sign in page
+  if (!req.session.asanaUserId && !req.session.googleUserId) {
+    return res.redirect("/");
+  }
+
+  // If no Google userId in session, redirect to Google sign in page
+  if (!req.session.googleUserId && req.session.asanaUserId) {
+    return res.redirect("/google/authorize/" + req.session.asanaUserId);
+  }
+
+  // If no Asana userId in session, redirect to Google sign in page
+  if (!req.session.asanaUserId && req.session.googleUserId) {
+    return res.redirect("/asana/authorize/" + req.session.googleUserId);
+  }
+
+  console.log("Asana User ID:" + req.session.asanaUserId);
+  console.log("Google User ID:" + req.session.googleUserId);
+  return next();
+};
+
+app.get("/tasks", handleSession, async (req, res) => {
+  // Retrieve an always fresh access token using userId
+  //console.log(req.body);
+  const asanaCredentials = await everyauth.getIdentity(
+    "asana",
+    req.session.asanaUserId
+  );
+
+  // Call Asana API
+  const asanaClient = asana.Client.create().useAccessToken(
+    asanaCredentials.accessToken
+  );
+
+  const me = await asanaClient.users.me();
+  const workspace = me.workspaces[0].gid;
+  const assignee = me.gid;
+
+  const tasks = await asanaClient.tasks.getTasks({ workspace, assignee });
+
+  const taskGIDs = tasks.data.map((tasks) => ({
+    taskGID: tasks.gid,
+  }));
+
+  const taskDetails = [];
+  for (const gid of taskGIDs) {
+    const task = await asanaClient.tasks.getTask(gid.taskGID);
+    taskDetails.push({
+      taskName: task.name,
+      taskNotes: task.notes,
+      taskDueDate: task.due_on,
+    });
+  }
+  // console.log({ TaskListData: taskDetails });
+
+  // Retrieve an always fresh access token using userId
+  const googleCredentials = await everyauth.getIdentity(
+    "google",
+    req.session.googleUserId
+  );
+
+  // Call Google API
+  const myAuth = new google.auth.OAuth2();
+  myAuth.setCredentials({ access_token: googleCredentials.accessToken });
+  google.options({ auth: myAuth });
+
+  const calendar = google.calendar({ version: "v3" });
+  const calendarEvents = await calendar.events.list({
+    auth: myAuth,
+    calendarId: "primary",
+    timeMin: new Date(),
+  });
+
+  const calendarEventsList = calendarEvents.data.items.map((calendarItem) => {
+    // Google Calendar API Returns Inconsistent Start Times
+    // So this is a workaround to handle the data and normalize it
+    var startTimeFromNow = "";
+    if (!calendarItem.start) {
+      startTimeFromNow = "";
+    } else if (calendarItem.start.dateTime) {
+      startTimeFromNow = moment(calendarItem.start.dateTime).fromNow();
+    } else if (calendarItem.start.date) {
+      startTimeFromNow = moment(calendarItem.start.date).fromNow();
+    }
+    return {
+      summary: calendarItem.summary,
+      description: calendarItem.description,
+      startTime: startTimeFromNow,
+    };
+  });
```

Comment:
Does this have anything to do with the Asana data?

Reply:
Negative.